lib/integrations/awsoidc/create_ec2ice.go

```diff
@@ -113,26 +133,41 @@ func CreateEC2ICE(ctx context.Context, clt CreateEC2ICEClient, req CreateEC2ICER
 		return nil, trace.Wrap(err)
 	}
 
-	createEC2ICEInput := &ec2.CreateInstanceConnectEndpointInput{
-		SubnetId:         &req.SubnetID,
-		SecurityGroupIds: req.SecurityGroupIDs,
-		TagSpecifications: []ec2types.TagSpecification{{
-			ResourceType: ec2types.ResourceTypeInstanceConnectEndpoint,
-			Tags:         req.ResourceCreationTags.ToEC2Tags(),
-		}},
-	}
-
-	ec2ICEndpoint, err := clt.CreateInstanceConnectEndpoint(ctx, createEC2ICEInput)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	endpointName := "unknown"
-	if ec2ICEndpoint.InstanceConnectEndpoint != nil {
-		endpointName = aws.ToString(ec2ICEndpoint.InstanceConnectEndpoint.InstanceConnectEndpointId)
+	createdEndpoints := make([]EC2ICEEndpoint, 0, len(req.Endpoints))
+	endpointNames := make([]string, 0, len(req.Endpoints))
+
+	for _, endpoint := range req.Endpoints {
+		createEC2ICEInput := &ec2.CreateInstanceConnectEndpointInput{
+			SubnetId:         &endpoint.SubnetID,
+			SecurityGroupIds: endpoint.SecurityGroupIDs,
+			TagSpecifications: []ec2types.TagSpecification{{
+				ResourceType: ec2types.ResourceTypeInstanceConnectEndpoint,
+				Tags:         req.ResourceCreationTags.ToEC2Tags(),
+			}},
+		}
+
+		ec2ICEndpoint, err := clt.CreateInstanceConnectEndpoint(ctx, createEC2ICEInput)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		// Sentinel value that indicates that the API returned a nil ec2ICEndpoint.InstanceConnectEndpoint.
+		// Very unlikely to happen.
+		endpointName := "unknown"
+		if ec2ICEndpoint.InstanceConnectEndpoint != nil {
+			endpointName = aws.ToString(ec2ICEndpoint.InstanceConnectEndpoint.InstanceConnectEndpointId)
+		}
+		endpointNames = append(endpointNames, endpointName)
+
+		createdEndpoints = append(createdEndpoints, EC2ICEEndpoint{
+			Name:             endpointName,
+			SubnetID:         endpoint.SubnetID,
+			SecurityGroupIDs: endpoint.SecurityGroupIDs,
+		})
 	}
 
 	return &CreateEC2ICEResponse{
-		Name: endpointName,
+		Name:             strings.Join(endpointNames, ","),
```

Comment:
Is this concatenated name still worth having?

Reply:
We have two sub-flows: either a single Endpoint or multiple Endpoints.
For the single Endpoint flow, we are still using the `Name` field.
So, in order for it to keep working, we must keep the field.